File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        s = list(s)
        if len(s) < 3:
            return ''.join(s)
        j = 2
        for i in range(2, len(s)):
            if s[i] != s[j - 1] or s[i] != s[j - 2]:
                s[j] = s[i]
                j += 1

        return ''.join(s[:j])

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        ios::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);
        
        if (s.size() < 3) return s;

        int j = 2;
        for (int i = 2; i < s.size(); ++i) {
            if (s[i] != s[j - 1] || s[i] != s[j - 2]) {
                s[j++] = s[i];
            }
        }
        s.resize(j);
        return s;
    }
};